### PR TITLE
[v4] feat(theme): Content width css variable for easier customization

### PR DESCRIPTION
**Pull request by elliotcourant**
This moves the content width that was used throughout nextra `max-w-[90rem]` to a css variable. This way the changing the value in the future should be easier, but it also makes it possible to customize the max width of the content for people using nextra without needing to overload a tailwind class in a hacky way.

## Why:

I have several screenshots in my docs that are a pretty consistent 1280x720, due to the max width of the content view being limited to `90rem` these images are always scaled down quite a bit. I was hoping to make the max width a bit wider for this reason but didn't see an easy way to do so without overloading the `_max-w-[90rem]` tailwind CSS class; which didn't seem like a good idea.

## What's being changed (if available, include any code snippets, screenshots, or gifs):

I've simply lifted the value passed to max width to a css variable similar to the other css variables already being managed by nextra. This way I can simply overwrite that css variable in my own styles.

## Check off the following:

- [x] I have reviewed my changes in staging, available via the **View
      deployment** link in this PR's timeline (this link will be available after
      opening the PR).

### CODE REVIEW

**Comment by elliotcourant** [comment]
> I don't think we can merge it, because there are people who rely on `_max-w-[90rem]` class
> 
> but in v4 we can do this feature
> 
> for now you can override this class in your CSS
> 
> ```css
> .max-w-\[90rem\] {
>     max-width: 90vw !important;
> }
> ```

Do you want me to redo this patch against the `v4-v2` branch? I'd be happy to.

**Comment by elliotcourant** [comment]
I've rebased onto the v4-v2 branch and made the changes there as well. Let me know if you have anymore thoughts!

File: packages/nextra-theme-docs/src/components/footer/index.tsx [hunk 1]
```diff
@@ -12,15 +12,15 @@ export const Footer: FC<ComponentProps<'footer'>> = ({
   return (
     <div className="x:bg-gray-100 x:pb-[env(safe-area-inset-bottom)] x:dark:bg-neutral-900 x:print:bg-transparent">
       <Switchers>
-        <div className="x:mx-auto x:flex x:max-w-[90rem] x:gap-2 x:py-2 x:px-4">
+        <div className="x:mx-auto x:flex x:max-w-[var(--nextra-content-width)] x:gap-2 x:py-2 x:px-4">
```
Comment (dimaMachina) [inline packages/nextra-theme-docs/src/components/footer/index.tsx:15]:
```suggestion
        <div className="x:mx-auto x:flex x:max-w-(--nextra-content-width) x:gap-2 x:py-2 x:px-4">
```

File: packages/nextra-theme-docs/src/components/footer/index.tsx [hunk 2]
```diff
@@ -12,15 +12,15 @@ export const Footer: FC<ComponentProps<'footer'>> = ({
...
           <LocaleSwitch />
           <ThemeSwitch />
         </div>
       </Switchers>
       <hr className="nextra-border" />
       <footer
         className={cn(
-          'x:mx-auto x:flex x:max-w-[90rem] x:justify-center x:py-12 x:text-gray-600 x:dark:text-gray-400 x:md:justify-start',
+          'x:mx-auto x:flex x:max-w-[var(--nextra-content-width)] x:justify-center x:py-12 x:text-gray-600 x:dark:text-gray-400 x:md:justify-start',
```
Comment (dimaMachina) [inline packages/nextra-theme-docs/src/components/footer/index.tsx:23]:
```suggestion
          'x:mx-auto x:flex x:max-w-(--nextra-content-width) x:justify-center x:py-12 x:text-gray-600 x:dark:text-gray-400 x:md:justify-start',
```

File: packages/nextra-theme-docs/src/components/navbar/index.tsx [hunk 3]
```diff
@@ -52,7 +52,7 @@ export const Navbar: FC<NavbarProps> = props => {
       />
       <nav
         style={{ height: 'var(--nextra-navbar-height)' }}
-        className="x:mx-auto x:flex x:max-w-[90rem] x:items-center x:justify-end x:gap-4 x:pl-[max(env(safe-area-inset-left),1.5rem)] x:pr-[max(env(safe-area-inset-right),1.5rem)]"
+        className="x:mx-auto x:flex x:max-w-[var(--nextra-content-width)] x:items-center x:justify-end x:gap-4 x:pl-[max(env(safe-area-inset-left),1.5rem)] x:pr-[max(env(safe-area-inset-right),1.5rem)]"
```
Comment (dimaMachina) [inline packages/nextra-theme-docs/src/components/navbar/index.tsx:55]:
```suggestion
        className="x:mx-auto x:flex x:max-w-(--nextra-content-width) x:items-center x:justify-end x:gap-4 x:pl-[max(env(safe-area-inset-left),1.5rem)] x:pr-[max(env(safe-area-inset-right),1.5rem)]"
```

File: packages/nextra-theme-docs/src/mdx-components/index.tsx [hunk 4]
```diff
@@ -89,7 +89,7 @@ const DEFAULT_COMPONENTS = getNextraMDXComponents({
       value: removeLinks(item.value)
     }))
     return (
-      <div className="x:mx-auto x:flex x:max-w-[90rem]">
+      <div className="x:mx-auto x:flex x:max-w-[var(--nextra-content-width)]">
```
Comment (dimaMachina) [inline packages/nextra-theme-docs/src/mdx-components/index.tsx:92]:
```suggestion
      <div className="x:mx-auto x:flex x:max-w-(--nextra-content-width)">
```

File: .changeset/afraid-foxes-deny.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'nextra-theme-docs': minor
+---
+
+Moved content width to a dedicated css variable.
```
Comment (dimaMachina) [inline .changeset/afraid-foxes-deny.md:5]:
```suggestion
content width could be configured via CSS `--nextra-content-width` variable
```

**Comment by elliotcourant** [comment]
Didn't know those shorthands existed in tailwind, thank you!

**Comment by dimaMachina** [comment]
> Didn't know those shorthands existed in tailwind, thank you!

they were added in tailwind v4

**Review by dimaMachina — APPROVED** [review]

**Pull request merged**